.changeset/goal-projection-cancel-race.md

```
@@ -0,0 +1,19 @@
+---
+'@mastra/core': patch
+'mastracode': patch
+---
+
+Fixed goals falsely appearing cancelled while their stored objective is still active.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)